src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/SessionMiddleware.cs

```diff
@@ -56,7 +57,11 @@ private async Task ManageStateAsync(HttpContextCore context, SessionAttribute me
         try
         {
             await _next(context);
-            await state.CommitAsync(context.RequestAborted);
+
+            // If the request is being redirected, the client may disconnect before we have a chance to commit, so we don't want to
+            // use HttpContext.RequestAborted as it will be disconnected
+            var token = context.GetAdapter().Response.IsRequestBeingRedirected ? CancellationToken.None : context.RequestAborted;
+            await state.CommitAsync(token);
```

Comment:
Consider not using a cancellation token here at all, or at least, using one that's not associated with the client. They're not really involved at this stage. As long as there's a basic timeout to prevent this from starving you should be fine.

Reply:
That works. I've updated it to have a 1 min timeout